src/components/FeedPortal/components/SideBar.jsx

```diff
@@ -1,62 +1,116 @@
 import React, { Fragment } from "react"
-import sidebarQuery from '../graphql/sidebarQuery';
+import PropTypes from "prop-types"
+import { gql } from "apollo-boost"
 import Request from "../../utilities/Request"
+import Loader from "../../Loader"
 
-const SidebarBtn = ({ lbl, active, team }) => (
-  <div className="sidebar-nav__btn-ctn">
-    {team ? <img className="sidebar-nav__btn-icon" src={require('../../../assets/team-icon.png')} alt="team-icon" /> : null}
-    <div className={`sidebar-nav__btn ${active ? "active" : null}`}>{lbl}</div>
-  </div>
-)
-
-const TeamLinks = teams => {
-  // TEMP REMOVE for people withr no teams !!!!!
-  if (!teams.length) {
-    teams = [{
-      title: "bears11",
-      cohort: { title: "voyage1" }
-    }, {
-      title: "dragons3",
-      cohort: { title: "voyage11" }
-    }]
+// -- QUERIES -- // 
+const sidebarHeaderQuery = gql`
+ query getSidebarUser{
+    user {
+      id
+      username
+      avatar
+    }
   }
+`
+const sidebarTeamsQuery = gql`
+  query getSidebarTeams{
+    user {
+      id
+      teams {
+        id
+        title
+        cohort {
+          id
+          title
+        }
+        tier {
+          id
+          level
+        }
+      }
+    }
+  }
+`
+
+// -- SIDEBAR HEADER -- // 
+const SidebarBtn = ({ lbl, active, team, toggleNewsFeed }) => {
+  return (
+    <div onClick={toggleNewsFeed} className="sidebar-nav__btn-ctn">
+      {team ? <img className="sidebar-nav__btn-icon" src={require('../../../assets/team-icon.png')} alt="team-icon" /> : null}
+      <div className={`sidebar-nav__btn ${active ? "active" : null}`}>{lbl}</div>
+    </div>
+  )
+}
+const SidebarHeader = ({ loading, data }) => {
+  if (loading) return <div style={{ height: "180.8px" }}><Loader style="medium" /></div>
```

Comment:
New task to make loader more flexible #125

Reply:
Yes, you're right :+1: